### PR TITLE
Update release notes script for milestones

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

Changed the release notes generation script to work for milestones in any repo rather than org level projects to align with the new release process.

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: scripts/release-notes/index.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    .filter(error => !!error);
+  if (errors.length) {
+    console.error(JSON.stringify(errors, null, 2));
+    throw new Error('Some issues are in an invalid state');
+  }
+  return issues;
+};
+
+const getIssues = async (milestoneId) => {
+  const response = await octokit.rest.issues.listForRepo({
```
Comment (craig-landry) [inline scripts/release-notes/index.js:86]: I was going to complain about getting all the issues without filtering to only include "Closed" issues. I see the prior version of the script did the same, so it's not a regression, just more a question: Do we include everything on purpose?

**Comment by craig-landry** [comment]
Only 1 minor change request, but it doesn't need to block if we just want this done (since it won't matter for quite a while). Approving so as not to block this in the mean time.

Also, when the release ticket describes running this script, it says `Ensure all issues are in the GH Project`, which should be changed to `Ensure all issues are marked with the expected milestone`. Please adjust https://github.com/medic/cht-core/issues/6943 and the template (I assume there is one?) that comes from.

**Review by craig-landry — APPROVED** [review]

File: scripts/release-notes/index.js (hunk 1 above)
Comment (garethbowen) [reply scripts/release-notes/index.js:86]: Yes we do. What it means is you can start drafting the release notes before all issues are closed. Obviously if you do this you need to update it if issues are added/removed but it's a useful starting point.

File: scripts/release-notes/index.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+];
+
+const PREFIXES_TO_IGNORE = [
+  'Type: Internal process',
+  'Won\'t fix:',
+  'Type: Investigation',
+];
+
+const getMilestone = async () => {
+  const response = await octokit.rest.issues.listMilestones({ owner: OWNER, repo: REPO_NAME });
```
[the remark this answers is not in the document]
Comment (garethbowen) [reply scripts/release-notes/index.js:52]: It turns out the octokit has a very handy "pagination" function which handles this for you, so I've added that. Thanks for the prod!

**Pull request merged**